actioncable/app/assets/javascripts/action_cable/connection.coffee

```diff
@@ -46,12 +51,21 @@ class ActionCable.Connection
     else
       @open()
 
+  getProtocol: ->
+    @webSocket?.protocol
+
+  isProtocolSupported: ->
+    @getProtocol() in protocols and @getProtocol() isnt unsupported_protocol
```

Comment:
The client isn't choosing the protocol, the server is. The client presents the server with a list of protocols it supports, the server chooses one which matches its own list of supported protocols, and sends that one (or none) back. If the browser client sees that none are sent back, it disconnects. That is all outside of our control.